pkg/registry/core/rest/storage_core.go

```diff
@@ -351,6 +352,37 @@ func (c *Config) newServiceIPAllocators() (registries rangeRegistries, primaryCl
 		if err != nil {
 			return rangeRegistries{}, nil, nil, nil, err
 		}
+		var bitmapAllocator ipallocator.Interface
+		if !utilfeature.DefaultFeatureGate.Enabled(features.DisableAllocatorDualWrite) {
+			bitmapAllocator, err = ipallocator.New(&serviceClusterIPRange, func(max int, rangeSpec string, offset int) (allocator.Interface, error) {
+				mem := allocator.NewAllocationMapWithOffset(max, rangeSpec, offset)
```

Comment:
I don't love that this is an exact dup of above - pull it into a func?

Reply:
is not an exact dupe, it has the logic to initialze the opaque object

```
	rangeRegistry.Range = serviceClusterIPRange.String()
				if len(rangeRegistry.ResourceVersion) == 0 {
					klog.Infof("kube-apiserver started with IP allocator and dual write enabled but bitmap allocator does not exist, recreating it ...")
					err := etcd.CreateOrUpdate(rangeRegistry)
					if err != nil {
						return nil, err
					}
				}
```
another benefit is that the diffs will be cleaner once we remove the feature gates, despite now is uglier to read